frontend/src/pages/article/article-page.tsx

```diff
@@ -19,8 +19,12 @@ const ArticlePage: React.FC = () => {
 
   const { id } = useParams();
 
+  const [initialDataLoad, setInitialDataLoad] = useState(false);
```

Comment:
I found that dataStatus is initially fulfilled here because the state is updated on the previous page when all articles are loaded. That's why, there was a bug with the Redirect More link (an immediate redirect occurred before the data was loaded). To fix it, I added a separate state variable to track whether the initial data loaded.